tests/test_makepyfile.py

```diff
@@ -0,0 +1,262 @@
+from encodings.aliases import aliases
+from pathlib import Path
+from textwrap import dedent
+from typing import Tuple
+
+import pytest
+
+pytest_plugins = ["pytester"]
+
+
+def enc(enc):
+    try:
+        _ = "abc".encode(enc)
+    except Exception:
+        return False
+
+    return True
+
+
+all_encodings = set(filter(enc, aliases.values()))
+
+
+@pytest.fixture(params=all_encodings)
+def encoding(request):
+    return request.param
+
+
+# UTF-8 Fixture (contains various Unicode characters)
+@pytest.fixture(
+    params=[
+        ("3", "4"),
+        ("☆", "★"),
+        ("1", "☀"),
+        ("✅", "💥"),
+    ],
+    ids=[
+        "numbers",
+        "stars",
+        "number-sun",
+        "check-explosion",
+    ],
+)
+def utf8_charset(request):
+    return request.param
+
+
+# CP-1252 Fixture (contains characters from Windows-1252 encoding)
+@pytest.fixture(
+    params=[
+        ("é", "ü"),
+        ("„", "“"),
+        ("¥", "ƒ"),
+        ("©", "®"),
+    ],
+    ids=[
+        "accented_letters_CP1252",
+        "quotes_CP1252",
+        "currency_symbols_CP1252",
+        "copyright_registered_CP1252",
+    ],
+)
+def cp1252_charset(request):
+    return request.param
+
+
+# ANSI (Strictly ASCII characters only)
+@pytest.fixture(
+    params=[
+        ("A", "B"),
+        ("1", "2"),
+        ("!", "?"),
+        ("@", "#"),
+    ],
+    ids=[
+        "letters_ANSI",
+        "numbers_ANSI",
+        "punctuation_ANSI",
+        "symbols_ANSI",
+    ],
+)
+def ansi_charset(request):
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        ("3", "4"),
+        ("A", "B"),
+        ("!", "?"),
+        ("☆", "☁"),
+        ("✅", "💥"),
+        ("é", "ü"),
+        ("Я", "ж"),  # Cyrillic characters
+        ("Ф", "Щ"),  # More Cyrillic
+        ("中", "国"),  # Chinese characters (zhong guo - China)
+        ("道", "德"),  # Chinese characters (dao de - way virtue)
+    ],
+    ids=[
+        "numbers",
+        "letters",
+        "punctuation",
+        "stars",
+        "check-explosion",
+        "accented_letters",
+        "cyrillic_basic",
+        "cyrillic_complex",
+        "chinese_country",
+        "chinese_concept",
```

Comment:
Why do these have to be tested separately? If your goal is to differentiate between traditional and simplified Chinese characters, there are better options. Also, if we start to go into such details, what about kanji and hiragana in Japanese? Let's not over-engineer the tests here.

Reply:
i just wanted to make sure not to only include latin chars, numbers and icons. 

so i asked the all mighty AI to create some data for cyrillic and chinese, i hope the are not offensive in any way.

Reply:
i changed my fixture [c8f55a2](https://github.com/scientific-python/pytest-doctestplus/pull/286/commits/c8f55a229184862952b95994231371aa7ebedf33)